src/ArrowFunctions.cpp

```diff
@@ -50,68 +50,127 @@ bool check_status_ok(arrow::Status status, char** errMsg) {
+    std::unique_ptr<parquet::arrow::FileReader> reader;
+    ARROWSTATUS_OK(parquet::arrow::OpenFile(infile, arrow::default_memory_pool(), &reader));
   
-  return reader -> parquet_reader() -> metadata() -> num_rows();
+    return reader -> parquet_reader() -> metadata() -> num_rows();
+  } catch (const std::exception& e) {
+    *errMsg = strdup(e.what());
+    return ARROWERROR;
+  }
 }
 
 int cpp_getType(const char* filename, const char* colname, char** errMsg) {
-  std::shared_ptr<arrow::io::ReadableFile> infile;
-  ARROWRESULT_OK(arrow::io::ReadableFile::Open(filename, arrow::default_memory_pool()),
-                 infile);
-
-  std::unique_ptr<parquet::arrow::FileReader> reader;
-  ARROWSTATUS_OK(parquet::arrow::OpenFile(infile, arrow::default_memory_pool(), &reader));
-
-  std::shared_ptr<arrow::Schema> sc;
-  std::shared_ptr<arrow::Schema>* out = &sc;
-  ARROWSTATUS_OK(reader->GetSchema(out));
-
-  int idx = sc -> GetFieldIndex(colname);
-  // Since this doesn't actually throw a Parquet error, we have to generate
-  // our own error message for this case
-  if(idx == -1) {
-    std::string fname(filename);
-    std::string dname(colname);
-    std::string msg = "Dataset: " + dname + " does not exist in file: " + filename; 
-    *errMsg = strdup(msg.c_str());
+  try {
+    std::shared_ptr<arrow::io::ReadableFile> infile;
+    ARROWRESULT_OK(arrow::io::ReadableFile::Open(filename, arrow::default_memory_pool()),
+                   infile);
+
+    std::unique_ptr<parquet::arrow::FileReader> reader;
+    ARROWSTATUS_OK(parquet::arrow::OpenFile(infile, arrow::default_memory_pool(), &reader));
+
+    std::shared_ptr<arrow::Schema> sc;
+    std::shared_ptr<arrow::Schema>* out = &sc;
+    ARROWSTATUS_OK(reader->GetSchema(out));
+
+    int idx = sc -> GetFieldIndex(colname);
+    // Since this doesn't actually throw a Parquet error, we have to generate
+    // our own error message for this case
+    if(idx == -1) {
+      std::string fname(filename);
+      std::string dname(colname);
+      std::string msg = "Dataset: " + dname + " does not exist in file: " + filename; 
+      *errMsg = strdup(msg.c_str());
+      return ARROWERROR;
+    }
+    auto myType = sc -> field(idx) -> type();
+
+    if(myType->id() == arrow::Type::INT64)
+      return ARROWINT64;
+    else if(myType->id() == arrow::Type::INT32)
+      return ARROWINT32;
+    else if(myType->id() == arrow::Type::UINT64)
+      return ARROWUINT64;
+    else if(myType->id() == arrow::Type::UINT32)
+      return ARROWUINT32;
+    else if(myType->id() == arrow::Type::TIMESTAMP)
+      return ARROWTIMESTAMP;
+    else if(myType->id() == arrow::Type::BOOL)
+      return ARROWBOOLEAN;
+    else if(myType->id() == arrow::Type::STRING)
+      return ARROWSTRING;
+    else if(myType->id() == arrow::Type::FLOAT)
+      return ARROWFLOAT;
+    else if(myType->id() == arrow::Type::DOUBLE)
+      return ARROWDOUBLE;
+    else // TODO: error type not supported
```

Comment:
Since error handling is being configured in this PR, should we go ahead and add the error here? Looking at the places where this is called, it looks like it will continue on returning `ARROWUNDEFINED` and it looks like actually producing/handling the error at this point would be more accurate.